### PR TITLE
Storage STG 95

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

**Review by nickliu-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareProperties.cs [hunk 1]
```diff
@@ -119,6 +119,24 @@ public class ShareProperties
         /// </summary>
         public bool? EnableSnapshotVirtualDirectoryAccess { get; internal set; }
 
+        /// <summary>
+        ///  Optional. Supported in version 2023-11-03 and above.  Only applicable for premium file storage accounts.
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareProperties.cs:123]: NIT - Is this supposed to be 2024-11-04 ? 

Same for the rest of the comments below

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareSetPropertiesOptions.cs [hunk 2]
```diff
@@ -30,6 +30,25 @@ public class ShareSetPropertiesOptions
         /// </summary>
 
         public bool? EnableSnapshotVirtualDirectoryAccess { get; set; }
+
+        /// <summary>
+        ///  Optional. Supported in version 2023-11-03 and above.  Only applicable for premium file storage accounts.
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareSetPropertiesOptions.cs:35]: same here

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs [hunk 3]
```diff
@@ -644,6 +656,18 @@ private void SetNameFieldsIfNull()
         /// Specifies whether the snapshot virtual directory should be accessible at the root of share mount point when NFS is enabled.
         /// If not specified, the default is true.
         /// </param>
+        /// <param name="enablePaidBursting">
+        ///  Optional. Supported in version 2023-11-03 and above.  Only applicable for premium file storage accounts.
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs:660]: same here

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs [hunk 4]
```diff
@@ -919,6 +964,18 @@ private void SetNameFieldsIfNull()
...
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs:968]: same here.

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs [hunk 5]
```diff
@@ -15,6 +15,7 @@
 using Azure.Storage.Sas;
 using Metadata = System.Collections.Generic.IDictionary<string, string>;
 using Azure.Storage.Common;
+using System.Net.Http.Headers;
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs:18]: Is this added by accident or is it being used in this file?

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareProperties.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareProperties.cs]: Fixed.

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareSetPropertiesOptions.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareSetPropertiesOptions.cs]: Fixed.

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs (hunk 4 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs]: Fixed.

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs (hunk 5 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs:18]: Removed.

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs]: Fixed.

**Pull request merged**